sys/arm64/arm64/exec_machdep.c

```diff
@@ -1009,7 +1009,6 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	tf->tf_x[8] = (uintcap_t)catcher;
 	tf->tf_sp = (uintcap_t)fp;
 #if __has_feature(capabilities)
-	tf->tf_x[9] = 0;
```

Comment:
Uh was this ever needed then? If it was needed before I don't see why it isn't any more?

Reply:
It was needed because trampolines checked `cheri_gettag(c9)` to decide whether to copy arguments over to the other stack. Now this check is no longer needed because the callee can directly access the arguments via `c9` if needed.

Reply:
Ok, but you haven't changed the trampoline yet, so it still needs c9 to always be set, which this patch very much breaks (not just for sendsig, but function calls in general), so won't this break c18n?

Reply:
Ah right. The `c18n` branch has a commit (https://github.com/CTSRD-CHERI/cheribsd/commit/1376d38e47b03cdc29e6db15b38d2967aa04bb46) that fixes RTLD, but this PR was cherry-picked from that branch. Let me try to cherry-pick that one in as well.

Reply:
Well now you have the entirety of your other c18n PR here, but most of it doesn't belong

Reply:
Right. The situation is that the c18n changes here depend on some previous commits (#1979), so we have to merge that PR first.

The changes in the PR and that one are almost entirely orthogonal.

Reply:
But doesn't it belong in the commit that changes the c18n code, rather than here?